src/app/features/ledger/utils/generic-ledger-utils.ts

```diff
@@ -48,3 +61,55 @@ export function prepareLedgerDeviceForAppFn<T extends () => Promise<unknown>>(co
     return app;
   };
 }
+
+async function getAppAndVersion(transport: Transport): Promise<{
+  name: string;
+  version: string;
+  flags: number | Buffer;
+}> {
+  const r = await transport.send(0xb0, 0x01, 0x00, 0x00);
+  let i = 0;
+  const format = r[i++];
+
+  if (format !== 1) {
+    throw new Error('getAppAndVersion: format not supported');
+  }
+
+  const nameLength = r[i++];
+  const name = r.subarray(i, (i += nameLength)).toString('ascii');
+  const versionLength = r[i++];
+  const version = r.subarray(i, (i += versionLength)).toString('ascii');
+  const flagLength = r[i++];
+  const flags = r.subarray(i, (i += flagLength));
+  return {
+    name,
+    version,
+    flags,
+  };
+}
```

Comment:
Isn't this code in the respective wrapper libraries?

Reply:
good point, i'll create a tmp bitcoin app based on that transport and just request the app name from that instance method